client/src/Components/Signup/index.jsx

```diff
@@ -1,11 +1,144 @@
-import React from 'react';
-import UnderConstruction from '../UnderConstruction';
-
-export default function SignUp() {
-  return (
-    <>
-      <h1>SIGN UP</h1>
-      <UnderConstruction />
-    </>
-  );
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import './style.css';
+import { Form, Button } from 'react-bootstrap';
+import signupValidation from './validation';
+
+export default class SignUp extends Component {
+  state = {
+    username: '',
+    email: '',
+    password: '',
+    confPassword: '',
+    errormsg: '',
+  };
+
+  handleClick = e => {
+    e.preventDefault();
+
+    const { username, password, email, confPassword } = this.state;
+    this.setState({ errormsg: '' });
+    signupValidation
+      .validate(
+        {
+          email,
+          password,
+          confPassword,
+          username,
+        },
+        { abortEarly: false }
+      )
+      .then(() => {
+        // fetch to back-end
+      })
+      .catch(({ inner }) => {
+        if (inner) {
+          const errors = inner.reduce(
+            (acc, item) => ({ ...acc, [item.path]: item.message }),
+            {}
+          );
+          this.setState({ errormsg: { ...errors } });
+        }
+      });
+  };
+
+  handleChange = ({ target: { value, name } }) => {
+    this.setState({ [name]: value });
```

Comment:
```suggestion
    this.setState(({ errormsg }) => {
      const newErrormsg = { ...errormsg };
      newErrormsg[name] = '';
      return { [name]: value, errormsg: newErrormsg };
    });
```

This way the error msg will disappear when the user starts typing